crates/api/src/v1/monitor.rs

```diff
@@ -0,0 +1,236 @@
+//! Types relating to the monitor API.
+
+use crate::Status;
+use serde::{de::Unexpected, Deserialize, Serialize, Serializer};
+use std::borrow::Cow;
+use thiserror::Error;
+use warg_crypto::hash::AnyHash;
+use warg_crypto::signing;
+use warg_protocol::registry::RegistryLen;
+
+/// Represents checkpoint verification response in one of the following states:
+#[derive(Serialize, Deserialize)]
+#[serde(tag = "state", rename_all = "camelCase")]
+#[allow(clippy::large_enum_variant)]
+pub enum CheckpointVerificationResponse {
+    /// The checkpoint is verified.
+    #[serde(rename_all = "camelCase")]
+    Verified,
+    /// The checkpoint is verified but has not verified the signature.
+    #[serde(rename_all = "camelCase")]
+    VerifiedButSignatureUnverified,
```

Comment:
Maybe merge these?

```suggestion
    Verified {
      #[serde(skip_serializing_if = "is_false")]
      signature_verified: bool,
    }
```

Some (most?) consumers won't care whether the signature has been verified or not.

Reply:
Walked Kyle thru the PR changes again this morning. He thought it would be better if we responded with "parallel" tracks for what was verified or unverified. The checkpoint (log length, map root and log root) and the signature (key ID, signature). And just using HTTP success response codes to indicate that the request succeeded but the client needs to interpret the response body for the verification results. Thoughts?

Reply:
Sure sounds fine.